pkg/cmd/util/clientcmd/factory_client_access.go

```diff
@@ -103,7 +103,7 @@ func DefaultClientConfig(flags *pflag.FlagSet) kclientcmd.ClientConfig {
 	cobra.MarkFlagFilename(flags, config.OpenShiftConfigFlagName)
 
 	// set our explicit defaults
-	defaultOverrides := &kclientcmd.ConfigOverrides{ClusterDefaults: kclientcmdapi.Cluster{Server: os.Getenv("KUBERNETES_MASTER")}}
+	defaultOverrides := &kclientcmd.ConfigOverrides{ClusterDefaults: kclientcmd.ClusterDefaults}
```

Comment:
@liggitt this matches what runs upstream. If you prefer your 44570, I have no objection, but please make sure it works here :-).